storm-server/src/main/java/org/apache/storm/scheduler/Cluster.java

```diff
@@ -80,6 +80,13 @@
     private final Map<String, Map<String, Double>> nodeToScheduledOffHeapNodeMemoryCache;   // node -> topologyId -> double
     private final Map<String, Set<WorkerSlot>> nodeToUsedSlotsCache;
     private final Map<String, NormalizedResourceRequest> totalResourcesPerNodeCache = new HashMap<>();
+    /**
+     * Snapshot of cluster total resources (cpu, memory, generic).
+     */
+    private final double totalCpuResource;
+    private final double totalMemoryResource;
+    private final Map<String, Double> totalGenericResources;
```

Comment:
These three variables nee to be part if copying from src in copy constructor and  `public Cluster(Cluster src, Topologies topologies)`

Reply:
Fixed

Reply:
If these variables are being used to keep running totals, should these be called " cacheTotalCpuResource" and so on. And the totals be incrementally updated in assign and unassign? If not running cache, then maybe just getTotalResouceCpu() == computeClusterCpuResource()?